src/org/omegat/util/EncodingDetector.java

```diff
@@ -54,17 +56,32 @@ public static String detectEncoding(File inFile) throws IOException {
      * @see <a href="https://github.com/amake/juniversalchardet">Fork</a>
      */
     public static String detectEncoding(InputStream stream) throws IOException {
-        UniversalDetector detector = new UniversalDetector(null);
```

Comment:
I'm not sure I like mixing this new encoding marker scheme in with `EncodingDetector`.

The approaches are very different, and I don't think it makes sense to do them in the same place, or with every file.

Instead, I think I would suggest that for glossary only (like in `GlossaryReaderTSV`) we do:

1. Have a separate detecting method that reads a single line of the file as ASCII and looks for the magic comment
2. If that fails, detect with `EncodingDetector` as we already do